test/ComposableCoW.stoploss.t.sol

```diff
@@ -54,57 +66,164 @@ contract ComposableCoWStopLossTest is BaseComposableCoWTest {
         stopLoss.getTradeableOrder(safe, address(0), bytes32(0), abi.encode(data), bytes(""));
     }
 
-    function test_strikePriceNotMet_fuzz(int256 sellTokenOraclePrice, int256 buyTokenOraclePrice, int256 strike)
+    function test_RevertStrikePriceNotMet_fuzz(int256 sellTokenOraclePrice, int256 buyTokenOraclePrice, int256 strike, uint256 currentTime, uint256 staleTime)
         public
     {
         vm.assume(buyTokenOraclePrice > 0);
         vm.assume(sellTokenOraclePrice > 0);
         vm.assume(strike > 0);
         vm.assume(sellTokenOraclePrice / buyTokenOraclePrice > strike);
+        vm.assume(currentTime > staleTime);
+
+        vm.warp(currentTime);
 
         StopLoss.Data memory data = StopLoss.Data({
-            sellToken: SELL_TOKEN,
-            buyToken: BUY_TOKEN,
-            sellTokenPriceOracle: mockOracle(SELL_ORACLE, sellTokenOraclePrice),
-            buyTokenPriceOracle: mockOracle(BUY_ORACLE, buyTokenOraclePrice),
+            sellToken: mockToken(SELL_TOKEN, DEFAULT_DECIMALS),
+            buyToken: mockToken(BUY_TOKEN, DEFAULT_DECIMALS),
+            sellTokenPriceOracle: mockOracle(SELL_ORACLE, sellTokenOraclePrice, block.timestamp, DEFAULT_DECIMALS),
+            buyTokenPriceOracle: mockOracle(BUY_ORACLE, buyTokenOraclePrice, block.timestamp, DEFAULT_DECIMALS),
             strike: strike,
             sellAmount: 1 ether,
             buyAmount: 1 ether,
             appData: APP_DATA,
             receiver: address(0x0),
             isSellOrder: false,
             isPartiallyFillable: false,
-            validityBucketSeconds: 15 minutes
+            validityBucketSeconds: 15 minutes,
+            heartBeatInterval: staleTime
+        });
+
+        vm.expectRevert(IConditionalOrder.OrderNotValid.selector);
+        stopLoss.getTradeableOrder(safe, address(0), bytes32(0), abi.encode(data), bytes(""));
+    }
+
+    function test_OracleNormalisesPrice_concrete() public {
```

Comment:
Could potentially make a fuzz test version here where decimals for the token and oracle are variable, if you want to be super fancy.

Reply:
Super fancy `fuzz` test included 💫